modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SFunction.java

```diff
@@ -128,6 +121,8 @@ void generateSignature(PainlessLookup painlessLookup) {
 
     @Override
     void analyze(ScriptRoot scriptRoot, Locals locals) {
+        this.settings = scriptRoot.getCompilerSettings();
```

Comment:
This is only used for `getMaxLoopCounter()`, consider just saving that setting.

Reply:
Same as the comment above.